### ISSUE TITLE
hdiutil: resize: failed. Invalid argument (22)

### ISSUE THREAD

**Issue by chrisspankroy** [verygenericname/SSHRD_Script#117]
My machine is now running macOS Ventura, but I could also reproduce the issue on Monterey.

Steps to reproduce:
1. Clone the repo
2. Run `./sshrd.sh 16.1`

The script will fail with an error `hdiutil: resize: failed. Invalid argument (22)`

This appears to be caused by this line in `sshrd.sh`:

`hdiutil resize -size 210MB work/ramdisk.dmg`

The README says that we can't resize the APFS DMGs on Linux machines, but it should work on macOS?

**Comment by verygenericname** [comment]
yes, i dont think apfs dmgs can be resized, umount /tmp/SSHRD then comment out that resize line and it should work

**Comment by chrisspankroy** [comment]
hm I get a bunch of `No space left on device` errors when it tries to copy files into that DMG, I'm guessing because we just commented out the line that makes it big enough to contain those files.

Could we make a new (bigger) APFS DMG and copy everything into that and just use that?

**Comment by verygenericname** [comment]
Not sure how to resize an apfs dmg

**Comment by chrisspankroy** [comment]
I opened #118 to fix this issue, but I can't actually get this project to boot on 16.1 to verify it works.

The screen lights up for a second and then shuts off. Does this project work on iOS 16.1?

**Comment by chrisspankroy** [comment]
Looking further at my boot issues, it looks like I can send the iBSS and iBEC normally (the screen is lit up) but the screen turns off as soon as I send `irecovery -c go` and the device goes to recovery mode.

What normally happens when `irecovery -c go` is run? Is the device supposed to go to recovery mode?

**Comment by chrisspankroy** [comment]
Ok I can confirm that #118 does work and closes this issue.

I'll open another issue for the side effects I'm seeing on iOS 16.

**Comment by ranzhie07** [comment]
> Ok I can confirm that #118 does work and closes this issue.
> 
> 
> 
> My issues with booting were resolved by using a newer version of irecovery

Can u help me im having error

![image](https://user-images.githubusercontent.com/13744581/201083419-68d4afb0-0d95-442f-bf31-3c879efdbcc6.jpeg)

**Comment by chrisspankroy** [comment]
Please don't double-post the same issue. The PR is the appropriate place to help you since that's what you're having an issue with

**Comment by ranzhie07** [comment]
> Please don't double-post the same issue. The PR is the appropriate place to help you since that's what you're having an issue with. This issue is unrelated to your issue.

I have new problem on ios16.1 
Iboot64patcher returning error cant patch ibss and ibec but on ios16 below patch is working

**Comment by ranzhie07** [comment]
> Please don't double-post the same issue. The PR is the appropriate place to help you since that's what you're having an issue with. This issue is unrelated to your issue.

Bro i already posted the output on issu #118

**Comment by ranzhie07** [comment]
> My machine is now running macOS Ventura, but I could also reproduce the issue on Monterey.
> 
> 
> 
> Steps to reproduce:
> 
> 1. Clone the repo
> 
> 2. Run `./sshrd.sh 16.1`
> 
> 
> 
> The script will fail with an error `hdiutil: resize: failed. Invalid argument (22)`
> 
> 
> 
> This appears to be caused by this line in `sshrd.sh`:
> 
> 
> 
> `hdiutil resize -size 210MB work/ramdisk.dmg`
> 
> 
> 
> The README says that we can't resize the APFS DMGs on Linux machines, but it should work on macOS?

I have comment 235 line same error
While running command
diskutil apfs resizeContainer $ourmountpoint 210m 

My os is bigsur v11.6.6

Error: 
![image](https://user-images.githubusercontent.com/13744581/201324115-fc5f9ab9-084c-4e34-8f05-ee78b7c89e5b.jpeg)

Heres the output of diskutil list
![image](https://user-images.githubusercontent.com/13744581/201324022-c8b760f9-cd95-4879-8254-8fce7a97fb42.jpeg)

**Comment by ranzhie07** [comment]
> Please don't double-post the same issue. The PR is the appropriate place to help you since that's what you're having an issue with. This issue is unrelated to your issue.

You have solution on problem in workaround?

**Comment by iam-theKid** [comment]
same here, ideas? thx
![image](https://user-images.githubusercontent.com/72942240/204780258-bcd005cd-1857-4829-a2f6-256a35f118e1.png)

**Comment by naiff654321** [comment]
> My machine is now running macOS Ventura, but I could also reproduce the issue on Monterey.
> 
> 
> 
> Steps to reproduce:
> 
> 1. Clone the repo
> 
> 2. Run `./sshrd.sh 16.1`
> 
> 
> 
> The script will fail with an error `hdiutil: resize: failed. Invalid argument (22)`
> 
> 
> 
> This appears to be caused by this line in `sshrd.sh`:
> 
> 
> 
> `hdiutil resize -size 210MB work/ramdisk.dmg`
> 
> 
> 
> The README says that we can't resize the APFS DMGs on Linux machines, but it should work on macOS?

**Comment by naiff654321** [comment]
palera1n.sh

**Comment by naiff654321** [comment]
> My machine is now running macOS Ventura, but I could also reproduce the issue on Monterey.
> 
> 
> 
> Steps to reproduce:
> 
> 1. Clone the repo
> 
> 2. Run `./sshrd.sh 16.1`
> 
> 
> 
> The script will fail with an error `hdiutil: resize: failed. Invalid argument (22)`
> 
> 
> 
> This appears to be caused by this line in `sshrd.sh`:
> 
> 
> 
> `hdiutil resize -size 210MB work/ramdisk.dmg`
> 
> 
> 
> The README says that we can't resize the APFS DMGs on Linux machines, but it should work on macOS?

**Comment by naiff654321** [comment]
name: Bug Report
description: File a bug report
title: "[Bug]: "
labels: ["bug", "triage"]
assignees:
  - octocat
body:
  - type: markdown
    attributes:
      value: |
        Thanks for taking the time to fill out this bug report!
  - type: input
    id: contact
    attributes:
      label: Contact Details
      description: How can we get in touch with you if we need more info?
      placeholder: ex. email@example.com
    validations:
      required: false
  - type: textarea
    id: what-happened
    attributes:
      label: What happened?
      description: Also tell us, what did you expect to happen?
      placeholder: Tell us what you see!
      value: "A bug happened!"
    validations:
      required: true
  - type: dropdown
    id: version
    attributes:
      label: Version
      description: What version of our software are you running?
      options:
        - 1.0.2 (Default)
        - 1.0.3 (Edge)
    validations:
      required: true
  - type: dropdown
    id: browsers
    attributes:
      label: What browsers are you seeing the problem on?
      multiple: true
      options:
        - Firefox
        - Chrome
        - Safari
        - Microsoft Edge
  - type: textarea
    id: logs
    attributes:
      label: Relevant log output
      description: Please copy and paste any relevant log output. This will be automatically formatted into code, so no need for backticks.
      render: shell
  - type: checkboxes
    id: terms
    attributes:
      label: Code of Conduct
      description: By submitting this issue, you agree to follow our [Code of Conduct](https://example.com)
      options:
        - label: I agree to follow this project's Code of Conduct
          required: true
